### ISSUE TITLE
Automated Checks Scorecard Feature Request

### ISSUE THREAD

**Issue by Dane2007** [enwikipedia-acc/waca#425]
In the concept image below, a backend system works through the request queue at a pre-determined interval (On Request Creation and then updates every 24h, 48h or 72h) to determine whether specific checks need user review.  A "Status Light" image is assigned based on certain criteria.  I have listed a proposed example criteria below.  A Green light would indicate no need to execute a check there as it is good, Yellow would indicate potential check needed if a red exists somewhere else; Red would indicate check needed in that area.

**Talk Page Checks:** Green if No talk page exists, Yellow if Content Exists, Red if Page exists with history but is blank currently.

**Local Contributions:** Green if No Contributions; Yellow if no contributions with in 6 months but contributions exist; Red if Contributions Exist with in 6 months

**Deleted Edits:** Green if No or low number of deleted edits exist; Red if Large number of deleted edits exist

**Global Contributions:** Green if No Contributions; Yellow if contributions older than 6 months exist; Red if Contributions Exist with in 6 months

**Local and Global Block Logs:** Green if No Blocks in Log; Red if Blocks in Log

**Active Local and Global Blocks:** Green if No Active Block; Red if Active Block Exist

**Abuse Filter Log:** Green if No Abuse Filter Entries; Yellow if older than 6 months; Red if with in 6 months.

![acc_conceptimage_autochecks](https://cloud.githubusercontent.com/assets/26441280/24486781/66144a26-14d1-11e7-8a9b-69ef607c397c.JPG)

**Comment by stwalkerster** [comment]
**tl;dr: this is complicated.**

For this to have a chance of being sanely implemented, this will need to depend on the job queue, which is being implemented in the OAuth creation / event management projects.

----

From a UI point of view, and because sometimes there's multiple IP addresses raised, I'm more tempted to try and combine this into the IP address buttons somehow. That way it's visually clearer which buttons are affected. We can also add this to the username / email buttons too

![image](https://cloud.githubusercontent.com/assets/722710/24505187/4a5b6200-1559-11e7-96b1-3d6847d91bfb.png)

**Let's also consider how this will be interpreted by users** - we don't want people to skip checks when there's some data there. Perhaps it's an idea to try and figure something out with yavascript, such that green checks are re-run on page-load. We don't have _any_ infrastructure like this at the moment though, so this would likely be significant work.

Data should be cached, as this will otherwise be a _significant_ performance impact on page-load. Caching also carries a much higher risk of being out-of-date, so we need to figure out a good cache invalidation time, and probably some way of regenerating that cache on expiry, otherwise we'll again be impacting page-load times. I think queuing items on the job queue is a good process to use here.

I also wonder if it's worth marking the "good" checks at all - I suspect the most value to be gained would be focusing on the failing checks first, as I'm not sure if we want to be able to skip certain checks from a policy perspective.

This also relates to the whois tasks - there was an idea to try integrate whois data into the tool itself - this would also reduce one of the external tools we rely on.

I'd like to ping @FastLizard4 for his opinion on this too. Technically, it'd be a nice challenge to implement, but I've got a fair number of those queued up right now.

**Comment by stwalkerster** [comment]
For the record, I have a whole bunch of code here external to the tool (in a [private repo](https://phabricator.stwalkerster.co.uk/diffusion/ACCREP/) due to embedded credentials) to do this.

I want to integrate it into the tool somehow, probably using the dynamic request lists from the event management code, but I don't want to start that until we have a baseline for newinternal in place.
